src/static/codeql/queries/cpp/openenclave/ECall-Args-IsOutsideEnclave.ql

```diff
@@ -13,49 +13,46 @@
 
 import cpp
 import semmle.code.cpp.Type
-import semmle.code.cpp.dataflow.TaintTracking
+import semmle.code.cpp.controlflow.IRGuards
+import semmle.code.cpp.dataflow.new.DataFlow
+import semmle.code.cpp.dataflow.new.TaintTracking
 import OpenEnclave
 
 /**
- * IsOutsideEnclaveBarrierGuard - A gaurd condition to check if a basic block is 
+ * isOutsideEnclaveBarrierGuardChecks - A gaurd condition to check if a basic block is
  * validated for envalve memory range protection by issuing a call to IsOutsideEnclave.
  */
-class IsOutsideEnclaveBarrierGuard extends DataFlow2::BarrierGuard {
-  IsOutsideEnclaveBarrierGuard() { this instanceof IsOutsideEnclaveFunctionCall }
-
-  override predicate checks(Expr checked, boolean isTrue) {
-    checked = this.(IsOutsideEnclaveFunctionCall).getArgument(0) and
-    isTrue = true
-  }
+predicate isOutsideEnclaveBarrierGuardChecks(IRGuardCondition g, Expr checked, boolean isTrue) {
+  checked = g.getUnconvertedResultExpression().(IsOutsideEnclaveFunctionCall).getArgument(0) and
```

Comment:
I am not sure if this line is correct, need review

Reply:
Think this would work? https://docs.github.com/en/code-security/codeql-cli/using-the-advanced-functionality-of-the-codeql-cli/testing-custom-queries

Reply:
Tested the code's behavior on demo code and its working as intended.